crates/orchestrator/src/jobs/da_job/mod.rs

```diff
@@ -198,30 +203,28 @@ pub fn convert_to_biguint(elements: Vec<FieldElement>) -> Vec<BigUint> {
     biguint_vec
 }
 
-fn data_to_blobs(blob_size: u64, block_data: Vec<BigUint>) -> Result<Vec<Vec<u8>>, JobError> {
+pub fn data_to_blobs(blob_size: u64, block_data: Vec<BigUint>) -> Result<Vec<Vec<u8>>, JobError> {
     // Validate blob size
     if blob_size < 32 {
         Err(DaError::InsufficientBlobSize { blob_size })?
     }
 
     let mut blobs: Vec<Vec<u8>> = Vec::new();
 
-    // Convert all FieldElements to bytes first
-    let mut bytes: Vec<u8> = block_data.iter().flat_map(|element| element.to_bytes_be().to_vec()).collect();
+    // Convert all BigUint to bytes more efficiently
+    let bytes: Vec<u8> = block_data.iter().flat_map(BigUint::to_bytes_be).collect();
```

Comment:
```suggestion
    let bytes: Vec<u8> = block_data.into_iter().flat_map(BigUint::to_bytes_be).collect();
```

whenever you're iterating over something that you won't use later, use `into_iter` as it moved the value instead of cloning

Reply:
got it.